### PR TITLE
Use the uploader event stream

### PR DESCRIPTION
**Pull request by rambler**
Wait for errors and completion. I think the problem here is that the process is exiting before AWS completes the operation.

### CODE REVIEW

File: lib/dynamo-backup.js [hunk 1]
```diff
@@ -55,8 +55,17 @@ DynamoBackup.prototype.backupTable = function (tableName, backupPath, callback)
         stream:     stream
     });
 
+    uploader.on('failed', function(err) {
+        console.log("Upload of table "+tableName+" failed with error: "+err);
```
Comment (dylanlingelbach) [inline lib/dynamo-backup.js]: @rambler - the `console.log` doesn't belong here.  An `error` event could be emitted from here, but I've tried to keep console logs out so people can choose how errors are displayed.

File: lib/dynamo-backup.js [hunk 2]
```diff
@@ -65,7 +74,9 @@ DynamoBackup.prototype.backupTable = function (tableName, backupPath, callback)
         },
         function(err) {
             stream.end();
-            callback(err);
+            if(err) {
```
Comment (dylanlingelbach) [inline lib/dynamo-backup.js]: This isn't quite right.  If `_copyTable` could return an error before we've even started streaming.  I think we should emit an error here, if an error was returned.

**Comment by dylanlingelbach** [comment]
@rambler - thanks for finding this bug and the PR.  Do you want to make the changes I suggested?  I can also make them if you prefer and get this merged in.

**Comment by rambler** [comment]
Sounds right to me. I've updated with the changes you propose.

**Comment by dylanlingelbach** [comment]
Thanks @rambler.  I appreciate you finding this and the fix!

Fix published in version 0.1.8.

**Pull request merged**